openstack/identity/v3/projects/results.go

```diff
@@ -0,0 +1,80 @@
+package projects
+
+import (
+	"github.com/mitchellh/mapstructure"
+	"github.com/rackspace/gophercloud"
+	"github.com/rackspace/gophercloud/pagination"
+)
+
+type commonResult struct {
+	gophercloud.Result
+}
+
+func (r commonResult) Extract() (*Project, error) {
+	if r.Err != nil {
+		return nil, r.Err
+	}
+
+	var response struct {
+		Project `json:"project"`
+	}
+
+	err := mapstructure.Decode(r.Body, &response)
+
+	return &response.Project, err
+}
+
+// CreateResult Project create result
+type CreateResult struct {
+	commonResult
+}
+
+// GetResult Project details result
+type GetResult struct {
+	commonResult
+}
+
+// UpdateResult Project update result
+type UpdateResult struct {
+	commonResult
+}
+
+// DeleteResult Project delete result
+type DeleteResult struct {
+	commonResult
+}
+
+// Project Project struct
+type Project struct {
```

Comment:
Let's move this to the top of the file so it's more pronounced. Also, can you add some descriptions for these fields?